datadog/dogstatsd/buffered_metrics.py

```diff
@@ -0,0 +1,70 @@
+import random
+from datadog.dogstatsd.metric_types import MetricType
+
+
+class BufferedMetric(object):
```

Comment:
for naming, I'm open to either sample or add_metric but I think we should keep it consistanted with the existing code in this repo, will follow up with Brian later to see if he has any thoughts.

For locks, when we added the other metric object types, [we waited to add the locks](https://github.com/DataDog/datadogpy/pull/837#discussion_r1658904919) and just kept the PR to the metric properties. I was thinking of doing the same here.

Reply:
Not a strong preference here. 

Slight preference for `sample` given the context in this case. Since each metric type inherits `BufferedMetric`, `sample` is an operation on the metric unlike the simple aggregator where `add_metric` adds a metric of a specific type. 

(could always go with `metric_sample` if you prefer 🤷 best of both)